src/snowflake/snowpark/mock/_stored_procedure.py

```diff
@@ -0,0 +1,174 @@
+    from typing import Iterable
+else:
+    from collections.abc import Iterable
+
+
+def sproc_types_are_compatible(x, y):
+    if isinstance(x, type(y)):
+        return True
+    elif isinstance(x, _IntegralType) and isinstance(y, _IntegralType):
+        return True
+    elif isinstance(x, _FractionalType) and isinstance(y, _FractionalType):
+        return True
+    return False
+
+
+class MockStoredProcedure(StoredProcedure):
+    def __call__(
+        self,
+        *args: Any,
+        session: Optional["snowflake.snowpark.session.Session"] = None,
+        statement_params: Optional[Dict[str, str]] = None,
+    ) -> Any:
+        args, session = self._validate_call(args, session)
+
+        # Unpack columns if passed
+        parsed_args = []
+        for arg, expected_type in zip(args, self._input_types):
+            if isinstance(arg, Column):
+                expr = arg._expression
+                if not sproc_types_are_compatible(expr.datatype, expected_type):
+                    raise ValueError(
+                        f"Unexpected type {expr.datatype} for sproc argument of type {expected_type}"
+                    )
+                parsed_args.append(
+                    calculate_expression(
+                        expr,
+                        ColumnEmulator(data=[None]),
+                        session._analyzer,
+                        {},
+                        keep_literal=True,
+                    )
+                )
+            else:
+                parsed_args.append(arg)
+
+        # import pdb; pdb.set_trace()
+
+        return self.func(session, *parsed_args)
+
+
+class MockStoredProcedureRegistration(StoredProcedureRegistration):
+    def __init__(self, *args, **kwargs) -> None:
+        super().__init__(*args, **kwargs)
+        self._registry: Dict[str, Callable] = dict()
+
+    def register_from_file(
+        self,
+        file_path: str,
+        func_name: str,
+        return_type: Optional[DataType] = None,
+        input_types: Optional[List[DataType]] = None,
+        name: Optional[Union[str, Iterable[str]]] = None,
+        is_permanent: bool = False,
+        stage_location: Optional[str] = None,
+        imports: Optional[List[Union[str, Tuple[str, str]]]] = None,
+        packages: Optional[List[Union[str, ModuleType]]] = None,
+        replace: bool = False,
+        if_not_exists: bool = False,
+        parallel: int = 4,
+        execute_as: typing.Literal["caller", "owner"] = "owner",
+        strict: bool = False,
+        external_access_integrations: Optional[List[str]] = None,
+        secrets: Optional[Dict[str, str]] = None,
+        *,
+        statement_params: Optional[Dict[str, str]] = None,
+        source_code_display: bool = True,
+        skip_upload_on_content_match: bool = False,
+    ) -> StoredProcedure:
+        raise NotImplementedError(
+            "[Local Testing] Registering stored procedure from file is not currently supported."
+        )
```

Comment:
let's use the telemetry not supported api.
also is this planned be supported by @sfc-gh-stan soon?

Reply:
Yea I'll have a separate PR that adds the support for imports.